drivers/modem/modem_shell.c

```diff
@@ -242,11 +245,114 @@ static int cmd_modem_info(const struct shell *shell, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_modem_sms_send(const struct shell *shell, size_t argc, char *argv[])
+{
+	struct ms_context *ms_ctx;
+	struct sms_out sms;
+	int ret= -1;
+
+	if (argc != 4) {
+		shell_fprintf(shell, SHELL_ERROR,
+			      "usage: modem sms send <modem index> <phone number> <message>\n");
+	}
+	else {
+		int modem_index = atoi(argv[1]);
+		ms_ctx = ms_context_from_id(modem_index);
+		if (ms_ctx == NULL) {
+			shell_fprintf(shell, SHELL_ERROR,
+				      "Couldn't open context for modem index %d\n", modem_index);
+			return -1;
```

Comment:
At least in the network shell, the `-ENOEXEC` is returned in case of an error. Don't know if it makes much difference here.